### PR TITLE
pin pycoco in hf translated model

### PR DESCRIPTION
**Pull request by novaturient95**
Temp fix to pip pycocotools which is causing deployment failure for mlflow converted models. Latest cython release caused this issue and a permanent change in being pushed in a dependent package.

### CODE REVIEW

File: assets/training/model_management/environments/model-management/context/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 azure-core=={{latest-pypi-version}}
 azure-ai-ml=={{latest-pypi-version}}
 azure-identity=={{latest-pypi-version}}
-azureml-evaluate-mlflow=={{latest-pypi-version}}
+azureml-evaluate-mlflow~={{latest-pypi-version}}
```
Comment (skasturi) [inline assets/training/model_management/environments/model-management/context/requirements.txt:4]: May be do this for all azureml packages. azureml-automl-core, azureml-telemetry?

File: assets/training/model_management/environments/model-management/context/requirements.txt [hunk 2]
```diff
@@ -10,3 +10,4 @@ wget=={{latest-pypi-version}}
 applicationinsights=={{latest-pypi-version}}
 azureml-automl-core=={{latest-pypi-version}}
 azureml-telemetry=={{latest-pypi-version}}
+pycocotools==2.0.4
```
Comment (skasturi) [inline assets/training/model_management/environments/model-management/context/requirements.txt:13]: This needs to move to the conda section. Build fails for this with pip. 

... or we need to upgrade to 2.0.6

File: assets/training/model_management/tests/dev_requirements.txt [hunk 3]
```diff
@@ -1,4 +1,5 @@
 # src
+pycocotools==2.0.4
```
Comment (skasturi) [inline assets/training/model_management/tests/dev_requirements.txt:2]: 2.0.4 doesn't compile with Cython 3.0 & fails during pip install. 

We could do either of the following: 
1. Move pycocotools=2.0.4 to conda section to pull from conda-forge
2. Change to pycocotools==2.0.6 and keep continuing to use pip

**Review by skasturi — CHANGES_REQUESTED** [review]
Please fix the env issues.

File: assets/training/model_management/environments/model-management/context/requirements.txt (hunk 2 above)
Comment (novaturient95) [reply assets/training/model_management/environments/model-management/context/requirements.txt]: updated to 2.0.6 in pip dep

File: assets/training/model_management/tests/dev_requirements.txt (hunk 3 above)
Comment (novaturient95) [reply assets/training/model_management/tests/dev_requirements.txt]: updated to 2.0.6 in pip dep

File: assets/training/model_management/src/azureml/model/mgmt/processors/transformers/convertors.py [hunk 4]
```diff
@@ -192,6 +194,52 @@ def _save(
             path=self._output_dir,
         )
 
+        # pin pycocotools==2.0.6
+        self._update_pip_dependencies({"pycocotools": "2.0.6"})
+
+    def _update_pip_dependencies(self, package_details):
```
Comment (skasturi) [inline assets/training/model_management/src/azureml/model/mgmt/processors/transformers/convertors.py:200]: Could we please add UTs for this?

File: assets/training/model_management/environments/model-management/context/requirements.txt [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+azureml-evaluate-mlflow~={{latest-pypi-version}}
 ffmpeg-python=={{latest-pypi-version}}
 langcodes=={{latest-pypi-version}}
 sentencepiece=={{latest-pypi-version}}
 accelerate=={{latest-pypi-version}}
 wget=={{latest-pypi-version}}
-applicationinsights=={{latest-pypi-version}}
-azureml-automl-core=={{latest-pypi-version}}
-azureml-telemetry=={{latest-pypi-version}}
+pycocotools==2.0.6
```
[the remark this answers is not in the document]
Comment (novaturient95) [reply assets/training/model_management/environments/model-management/context/requirements.txt:12]: Done

File: assets/training/model_management/src/azureml/model/mgmt/processors/transformers/convertors.py (hunk 4 above)
Comment (novaturient95) [reply assets/training/model_management/src/azureml/model/mgmt/processors/transformers/convertors.py]: Added

**Review by skasturi — APPROVED** [review]

**Review by vizhur — APPROVED** [review]

**Pull request merged**